beacon_node/beacon_chain/src/merge_readiness.rs

```diff
@@ -145,18 +152,20 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
             if let Err(e) = el.exchange_transition_configuration(&self.spec).await {
                 // The EL was either unreachable, responded with an error or has a different
                 // configuration.
-                return MergeReadiness::ExchangeTransitionConfigurationFailed(e);
+                return MergeReadiness::ExchangeTransitionConfigurationFailed {
+                    error: format!(
+                        "Error on exchange_transition_configuration endpoint {:?}",
+                        e
+                    ),
+                };
             }
 
             if !el.is_synced_for_notifier().await {
                 // The EL is not synced.
                 return MergeReadiness::NotSynced;
             }
             let params = MergeConfig::from_chainspec(&self.spec);
-            let current_difficulty = el
-                .get_current_difficulty()
-                .await
-                .map_err(|_| "Failed to get current difficulty from execution node".to_string());
+            let current_difficulty = el.get_current_difficulty().await.ok();
```

Comment:
If the `get_current_difficulty` call errors should we still be returning `Ready`? I'm tempted to say we should return a different variant

Reply:
If that call failed after all the previous checks, that would mostly mean an intermittent json rpc failure. If the EE is out for longer, then the next time anyone calls the function, it would return a EE offline error when doing the `exchange_transition_configuration`. 
Happy to do another error type for this if that's clearer though.

Reply:
Good point, let's merge as-is and we can revisit this if we want stronger guarantees on the API in future